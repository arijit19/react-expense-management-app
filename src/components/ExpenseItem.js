import React from 'react';
import moment from 'moment';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

const ExpenseItem = ({ id, description, amount, createdAt})=> (
    <div>
        <Link to={`/edit/:${id}`} className='list-item'>
            <div>
                <h3 className='list-item__title'>{description}</h3>
                <span className='list-item__sub-title'>{moment(createdAt).format('Do MMM YYYY')}</span>
            </div>
            <h3 className='list-item__data'>{numeral(amount).format('$0,0.00')}</h3>
        </Link>
    </div>
);

export default ExpenseItem;