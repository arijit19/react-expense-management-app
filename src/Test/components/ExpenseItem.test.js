import React from 'react';
import {shallow} from 'enzyme';

import ExpenseItem from '../../components/ExpenseItem.js'
import expenses from '../fixtures/Expense.js';

test('Test Expense item component', ()=>{
    const wrapper = shallow(<ExpenseItem key={expenses[0].id} {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot();
})