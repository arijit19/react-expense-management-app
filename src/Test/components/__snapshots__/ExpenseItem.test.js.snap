// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Expense item component 1`] = `
<div>
  <Link
    className="list-item"
    to="/edit/:1"
  >
    <div>
      <h3
        className="list-item__title"
      >
        Rent
      </h3>
      <span
        className="list-item__sub-title"
      >
        1st Jan 1970
      </span>
    </div>
    <h3
      className="list-item__data"
    >
      $20,000.00
    </h3>
  </Link>
</div>
`;
